deploy/examples/csi/rbd/storageclass-ec.yaml

```
@@ -62,8 +62,11 @@ parameters:
   # RBD image format. Defaults to "2".
   imageFormat: "2"
 
-  # RBD image features. Available for imageFormat: "2". CSI RBD currently supports only `layering` feature.
-  imageFeatures: layering
+  # RBD image features, equivalent to OR'd bitfield value: 63
+  # Available for imageFormat: "2". Older releases of CSI RBD
+  # support only the `layering` feature.  The Linux kernel (KRBD) supports the
+  # full feature complement as of 5.4
+  imageFeatures: layering,fast-diff,object-map,deep-flatten,exclusive-lock
```

```suggestion
  imageFeatures: layering,fast-diff,object-map,deep-flatten,exclusive-lock
```

I'd prefer we have only layering by default and list this alongside in a comment, so user can add it as required.
Most users will still be on lower kernel, they will have to edit this when they get an error after creating a PVC. 
This will be very inconvenient according to me.

---

I think today most users will be a kernel that's new enough ; users of older installations will still have the old script.

---

Looks like the other storage classes default to `layering`, and just the EC storage class defaults to the newer kernel features? Seems like we should be consistent with the defaults. 
To @Rakshith-R's point, the consequence of trying to run the defaults is painful since PVs will fail if you have an older kernel and don't know to change this. With the `layering` default, everything still works even if not optimally. We don't know how many users are on older vs newer kernels, but I'd feel ok going with the newer default if we felt like 95% of users are on the newer kernel. It still feels like we are not there. 
What if we add a note in the [Prerequisites doc](https://rook.github.io/docs/rook/latest/Getting-Started/Prerequisites/prerequisites/#kernel) that they should update the storage class if they have a 5.4+ kernel?

---

Fair enough.  I've tweaked accordingly.  PTAL.